WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasDisplayButton.lua

```diff
@@ -436,57 +428,56 @@ local methods = {
             end
         end
 
-        function self.callbacks.OnDragStart(_, _)
+        function self.callbacks.OnDragStart(frame, _)
             if WeakAuras.IsImporting() or not data.parent then return end;
 
-            data.dragging = true
+            local x, y = GetCursorPosition()
+            local scale = UIParent:GetScale()
 
-            self.frame:StartMoving()
-            self.frame:SetUserPlaced(false)
+            frame.isDragged = true
+            frame:StartMoving()
+            frame:ClearAllPoints()
+            frame:SetPoint("Center", UIParent, "BOTTOMLEFT", x/scale, y/scale)
+
+            WeakAuras:SortDisplayButtons() -- find a better way to trigger a DoLayout call on the scroll frame
         end
 
-        function self.callbacks.OnDragStop(_)
-            if data.dragging then
-                self.frame:StopMovingOrSizing()
+        function self.callbacks.OnDragStop(frame)
```

Comment:
You anchor to the hoovered element. Also you are updating on each frame anyway. As to how create a frame with a texture. That's something like this:

  local dragIndicator = CreateFrame("FRAME", "WeakAurasDragIndicator");
  dragIndicator:SetHeight(10); -- TODO depends
  dragIndicator:SetFrameStrata("FULLSCREEN"); -- above settings dialog
  
  local texture = dragIndicator:CreateTexture(nil, "BACKGROUND");
  texture:SetAllPoints(dragIndicator);
  texture:SetTexture("a path");

and then in once you have the frame to which you want to anchor:
dragIndicator:ClearAllPoints();
dragIndicator:SetPoint("TOPLEFT", anchor, "BOTTOMLEFT", xoffset, yoffset);
dragIndicator:SetPoint("TOPRIGHT", anchor, "BOTTOMRIGHT", xoffset, yoffset);